plugins/otter-pro/inc/plugins/class-block-conditions.php

```diff
@@ -566,6 +596,122 @@ public function has_products( $products ) {
 		return $bought;
 	}
 
+	/**
+	 * Check based on WooCommerce product category.
+	 *
+	 * @param array $categories IDs of Categories.
+	 *
+	 * @since  2.7.0
+	 * @access public
+	 */
+	public function has_product_category( $categories ) {
+		$in_category = false;
+
+		global $product;
+
+		if ( ! $product ) {
```

Comment:
Instead of `! $product`, it would be better to check if `$product` is actually a WooCommerce product object. Or at the very least, check if it has the `get_id` method that you use below. This will make the code more robust and prevent potential bugs in the future.